src/apps/answers/service.py

```diff
@@ -276,6 +276,34 @@ async def _create_answer(self, applet_answer: AppletAnswerCreate) -> AnswerSchem
         )
         return answer
 
+    async def validate_take_now(
+        self, applet_id: uuid.UUID, target_subject_id: uuid.UUID | None, source_subject_id: uuid.UUID | None
+    ):
+        assert self.user_id
+        subject_crud = SubjectsCrud(self.session)
+
+        respondent_subject = await subject_crud.get_user_subject(user_id=self.user_id, applet_id=applet_id)
+        if not respondent_subject or not respondent_subject.soft_exists():
+            raise ValidationError("Respondent subject not found")
+
+        target_subject = None
+        source_subject = None
+
+        if target_subject_id:
+            target_subject = await subject_crud.get_by_id(target_subject_id)
+            if not target_subject or not target_subject.soft_exists() or target_subject.applet_id != applet_id:
+                raise ValidationError(f"Subject {target_subject_id} not found")
+
+        if source_subject_id:
+            source_subject = await subject_crud.get_by_id(source_subject_id)
+            if not source_subject or not source_subject.soft_exists() or source_subject.applet_id != applet_id:
+                raise ValidationError(f"Subject {source_subject_id} not found")
+
+        if target_subject and source_subject and target_subject_id != source_subject_id:
+            relation = await SubjectsCrud(self.session).get_relation(source_subject.id, target_subject.id)
+            if not relation:
+                raise ValidationError("Source and Target relation not found")
```

Comment:
I don't think we're meant to be doing this validation yet. Can you clarify with @mbanting if this is supposed to be part of the acceptance criteria?

Reply:
I apologize @rcmerlo, the AC does include

> or the respondent is not a Team Admin but has a relationship with the target

because this ticket was originally scheduled for Release 2, when assignments and/or connections were potentially in Scope. Since we are doing this in Release 1, and since this ticket was created we may not even be doing assignments in R2 anymore anyways, the AC should have been updated to remove this line. 

Although it exists, we're not using the `subject_relations` table yet either.  I don't see the endpoint`/subjects/{subject_id}/relations/{source_subject_id}` being called by the admin panel or web app yet, and the table is empty in my local DB as well. 

So please go ahead and remove this, and I'll update the ticket's AC as well.

Reply:
removed